src/pages/task/TestsTable.tsx

```diff
@@ -42,23 +47,51 @@ export const TestsTable: React.FC = () => {
     } else if (!validInitialQueryParams) {
       const statuses = parsed[RequiredQueryParams.Statuses];
       setValidInitialQueryParams({
-        initialCategory: parsed[RequiredQueryParams.Category],
+        initialCategory: (
+          parsed[RequiredQueryParams.Category] || ""
+        ).toString(),
         initialSort: parsed[RequiredQueryParams.Sort],
         initialStatuses: (Array.isArray(statuses)
           ? statuses
           : [statuses]
-        ).filter(v => v && v !== TestStatus.All)
+        ).filter(v => v && v !== TestStatus.All),
+        initialTestName: testName
       });
     }
-  }, [search, pathname, replace, validInitialQueryParams]);
+  }, [search, pathname, replace, validInitialQueryParams, parsed, testName]);
 
   if (!validInitialQueryParams) {
     return null;
   }
+
+  const onSearch = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    parsed[RequiredQueryParams.TestName] = e.target.value;
+    const nextQueryParams = queryString.stringify(parsed);
+    replace(`${pathname}?${nextQueryParams}`);
+  };
+
   return (
     <>
-      <StatusSelector />
+      <FiltersWrapper>
+        <StyledInput
+          placeholder="Search Test Names"
+          onChange={onSearch}
```

Comment:
Should we debounce this input? Testresult queries can be expensive, so it'd be nice to not perform the search if it's not done